hooks/useFetchProjects.ts

```diff
@@ -94,7 +126,52 @@ const useFetchProjects = () => {
     fetchProjects();
   }, [projects, setProjects]);
 
-  return { projects, loading };
+  // handle tags extraction
+  useEffect(() => {
+    const newTags = extractTagsFromProjects(projects?.items?.items);
+    setTags(newTags);
+  }, [projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    const projectsToDisplay = filterProjectsByTags(
+      projects?.items?.items,
+      tagToFilterBy
+    );
+    setFilteredProjects(projectsToDisplay);
+  }, [tags, tagToFilterBy, projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    if (tags?.length) {
+      const newFiteredTags = structuredClone(tagToFilterBy);
+      tags.forEach(tag => {
+        const existingTag = newFiteredTags.find(t => t.name === tag);
+        if (!existingTag) {
+          newFiteredTags.push({ name: tag, isActive: true });
+        } else {
+          existingTag.isActive = !existingTag.isActive;
+        }
+      });
+      setTagToFilterBy(n => newFiteredTags);
+    }
```

Comment:
Will it work like this `setTagToFilterBy(newFiteredTags);` ? You are not using the `n` in the arrow function.

Reply:
Whenever we use the previous state to update the state, we should use the callback function.
Since some of these changes can be invoked in rapid succession this is the recommended approach.